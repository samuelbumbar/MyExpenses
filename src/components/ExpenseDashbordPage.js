import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummary';

const ExpenseDashbordPage = () => (
    <div>
        <ExpensesSummary />
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default ExpenseDashbordPage;