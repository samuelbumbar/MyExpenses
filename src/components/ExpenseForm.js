import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';

// const date = new Date();
const now = moment();
console.log(now.format('MMM Do, YYYY'));

export default class ExpenseForm extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            description: props.expense ? props.expense.description : '',
            note: props.expense ? props.expense.note : '',
            amount: props.expense ? (props.expense.amount / 100).toString() : '',
            createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
            calendarFocused: false,
            error: ''
        };
    }
    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState(() => ({ description }));
    };
    onNoteChange = (e) => {
        e.persist();    // if we want to use 'e' inside of a callback we have to call this function first
        this.setState(() => ({ note: e.target.value }));
    };
    onAmountChange = (e) => {
        const amount = e.target.value;
        if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/))
            this.setState(() => ({ amount }));
    };
    onDateChange = (createdAt) => {
        if (createdAt)  // This prevents the user from being able to clear the value
            this.setState(() => ({ createdAt }));
        // else    // If the date value is cleared, go back to the current date
        //     this.setState(() => ({ createdAt: moment() }));
    };
    onFocusChange = ({ focused }) => {
        this.setState(() => ({ calendarFocused: focused }));
    };
    onSubmit = (e) => {
        e.preventDefault();

        if (!this.state.description || !this.state.amount) {
            // Set error state equal to 'Please provide description and amount.'
            this.setState(() => ({ error: 'Please provide description and amount.' }));
        } else {
            // Clear the error
            this.setState(() => ({ error: '' }));

            this.props.onSubmit({
                description: this.state.description,
                amount: parseFloat(this.state.amount, 10) * 100,
                createdAt: this.state.createdAt.valueOf(),
                note: this.state.note
            });
        }
    };
    render() {
        return (
            <form className="form" onSubmit={this.onSubmit}>
                {this.state.error && <p className="form__error">{this.state.error}</p>}
                <input
                    type="text"
                    placeholder="Description"
                    autoFocus
                    className="text-input"
                    value={this.state.description}
                    onChange={this.onDescriptionChange}
                />
                <input
                    type="text"
                    placeholder="Amount"
                    className="text-input"
                    value={this.state.amount}
                    onChange={this.onAmountChange}
                />
                <SingleDatePicker
                    date={this.state.createdAt}             // sets where the date value is get from
                    onDateChange={this.onDateChange}        // sets the handler when setting a new value to the calendar
                    focused={this.state.calendarFocused}    // checks if the calendar should be displayed
                    onFocusChange={this.onFocusChange}      // changes the focus value if it changes
                    numberOfMonths={1}                      // sets the number of months to be displayed at the same time
                    isOutsideRange={(day) => false}         // disables the selected day verification
                />
                <textarea
                    placeholder="Add a note for your expense (optional)"
                    className="textarea"
                    value={this.state.note}
                    onChange={this.onNoteChange}
                />
                <div>
                    <button className="button">Save Expense</button>
                </div>
            </form>
        );
    }
};
