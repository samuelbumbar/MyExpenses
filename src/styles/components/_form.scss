.form {
    margin: 0;
    display: flex;
    flex-direction: column;
    >* {
        margin-bottom: $m-size;
    }
}

.form__error {
    margin: 0 0 $m-size 0;
    font-style: italic;
}