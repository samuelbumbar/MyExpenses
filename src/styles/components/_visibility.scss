.show-for-desktop {
  @media (max-width: $desktop-breakpoint) {
    display: none;
  }
}

.show-for-mobile {
  @media (min-width: $desktop-breakpoint - .01rem) {
    display: none;
  }
}

// .show-for-desktop {
//   display: none;
// }

// .show-for-mobile {
//   display: block;
// }

// @media (min-width: $desktop-breakpoint) {
//   .show-for-desktop {
//       display: block;
//   }
//   .show-for-mobile {
//       display: none;
//   }
// }