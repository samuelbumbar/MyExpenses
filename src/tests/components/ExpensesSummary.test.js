import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpensesSummary with one expense', () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={777} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={7} expensesTotal={114195} />);
  expect(wrapper).toMatchSnapshot();
});

// test('should render ExpensesSummary with one expense', () => {
//   const wrapper = shallow(<ExpensesSummary expenses={[expenses[0]]} />);
//   expect(wrapper).toMatchSnapshot();
// });

// test('should render ExpensesSummary with multiple expenses', () => {
//   const wrapper = shallow(<ExpensesSummary expenses={expenses} />);
//   expect(wrapper).toMatchSnapshot();
// });