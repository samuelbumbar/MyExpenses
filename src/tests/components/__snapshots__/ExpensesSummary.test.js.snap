// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        7
      </span>
       
      expenses
       totalling 
      <span>
        $1,141.95
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render ExpensesSummary with one expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        $7.77
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
